src/sudoers/mod.rs

```diff
@@ -81,6 +84,47 @@ impl Sudoers {
         Ok(analyze(path.as_ref(), sudoers))
     }
 
+    fn specify_host_and_user<User: UnixUser + PartialEq<User>>(
+        &mut self,
+        hostname: &system::Hostname,
+        user: &User,
+    ) {
+        let host_customisers = std::mem::take(&mut self.customisers.host);
+        let user_customisers = std::mem::take(&mut self.customisers.user);
+        specialise_setting(
+            &mut self.settings,
+            host_customisers,
+            &match_token(hostname),
+            &self.aliases.host,
+        );
+        specialise_setting(
+            &mut self.settings,
+            user_customisers,
+            &match_user(user),
+            &self.aliases.user,
+        );
+    }
+
+    fn specify_runas<User: UnixUser + PartialEq<User>>(&mut self, run_as: &User) {
```

Comment:
Are `specify_host_and_user` and `specify_runas` ever called separately? If not maybe merge them?